### ISSUE TITLE
Kubeflow on GKE not using the default gcp zone configured in the cloud shell.

### ISSUE THREAD

**Issue by dformoso** [kubeflow/kubeflow#1914]
Hi all,

I'm trying to install Kubeflow 0.3.1, and changing the default zone to us-central1-a. I am located in Sydney, Australia, and somehow Kubeflow picks up my location and uses that instead of us-central1-a.

This is how I run my setup:

# Run everything from container
`docker container run -it -d \
  --restart=always \
  --name google_sh \
  --hostname google_sh \
  google/cloud-sdk:latest

docker container exec -it google_sh bash

gcloud auth login
# Set Environment
export PROJECT=data-science-on-kubs
export REGION=us-central1
export ZONE=us-central1-a
gcloud config set project ${PROJECT}
gcloud config set compute/region ${REGION}
gcloud config set compute/zone ${ZONE}
gcloud config list

# Install ksonnet
export KS_VER=0.12.0
export KS_PKG=ks_${KS_VER}_linux_amd64
apt-get update && apt-get install wget -y
wget -O /tmp/${KS_PKG}.tar.gz https://github.com/ksonnet/ksonnet/releases/download/v${KS_VER}/${KS_PKG}.tar.gz --no-check-certificate
mkdir -p ${HOME}/bin
tar -xvf /tmp/$KS_PKG.tar.gz -C ${HOME}/bin
export PATH=$PATH:${HOME}/bin/$KS_PKG

# Install Kubeflow
export KUBEFLOW_REPO=kubeflow_dir
mkdir ${KUBEFLOW_REPO}
cd ${KUBEFLOW_REPO}
export KUBEFLOW_TAG=v0.3.1
curl https://raw.githubusercontent.com/kubeflow/kubeflow/${KUBEFLOW_TAG}/scripts/download.sh | bash

# Set up and deploy Kubeflow:
export KFAPP=myapp
/${KUBEFLOW_REPO}/scripts/kfctl.sh init ${KFAPP} --platform gcp --project ${PROJECT}
cd ${KFAPP}
/${KUBEFLOW_REPO}/scripts/kfctl.sh generate platform
/${KUBEFLOW_REPO}/scripts/kfctl.sh apply platform
`

# This is the error I get on apply:
```root@google_sh:/kubeflow_dir/myapp# /${KUBEFLOW_REPO}/scripts/kfctl.sh apply platform
+ COMMAND=apply
+ WHAT=platform
+ ENV_FILE=env.sh
+ SKIP_INIT_PROJECT=false
+ MIN_CLUSTER_VERSION=1.10.7-gke.2
+++ dirname /kubeflow_dir/scripts/kfctl.sh
++ cd /kubeflow_dir/scripts
++ pwd
+ DIR=/kubeflow_dir/scripts
+ source /kubeflow_dir/scripts/util.sh
+ source /kubeflow_dir/scripts/gke/util.sh
++ set -xe
+ source /kubeflow_dir/scripts/util-minikube.sh
++ RED='\033[0;31m'
++ GREEN='\033[0;32m'
++ YELLOW='\033[0;33m'
++ NC='\033[0m'
++ MOUNT_LOCAL=false
+ '[' apply == init ']'
+ source env.sh
++ PLATFORM=gcp
++ KUBEFLOW_REPO=kubeflow_dir
++ KUBEFLOW_VERSION=master
++ KUBEFLOW_KS_DIR=/kubeflow_dir/myapp/ks_app
++ KUBEFLOW_DOCKER_REGISTRY=
++ K8S_NAMESPACE=kubeflow
++ KUBEFLOW_CLOUD=gke
++ PROJECT=data-science-on-kubs
++ DEPLOYMENT_NAME=myapp
++ KUBEFLOW_DM_DIR=/kubeflow_dir/myapp/gcp_config
++ KUBEFLOW_SECRETS_DIR=/kubeflow_dir/myapp/secrets
++ KUBEFLOW_K8S_MANIFESTS_DIR=/kubeflow_dir/myapp/k8s_specs
++ KUBEFLOW_K8S_CONTEXT=myapp
++ ZONE=us-central1-a
++ EMAIL=formoso@google.com
++ KUBEFLOW_IP_NAME=myapp-ip
++ KUBEFLOW_ENDPOINT_NAME=myapp
++ KUBEFLOW_HOSTNAME=myapp.endpoints.data-science-on-kubs.cloud.goog
++ CONFIG_FILE=cluster-kubeflow.yaml
++ PROJECT_NUMBER=1067425295876
++ CLUSTER_VERSION=1.10.9-gke.0
+ '[' -z apply ']'
+ '[' -z platform ']'
+ check_install ks
+ which ks
+ check_install kubectl
+ which kubectl
+ source env.sh
++ PLATFORM=gcp
++ KUBEFLOW_REPO=kubeflow_dir
++ KUBEFLOW_VERSION=master
++ KUBEFLOW_KS_DIR=/kubeflow_dir/myapp/ks_app
++ KUBEFLOW_DOCKER_REGISTRY=
++ K8S_NAMESPACE=kubeflow
++ KUBEFLOW_CLOUD=gke
++ PROJECT=data-science-on-kubs
++ DEPLOYMENT_NAME=myapp
++ KUBEFLOW_DM_DIR=/kubeflow_dir/myapp/gcp_config
++ KUBEFLOW_SECRETS_DIR=/kubeflow_dir/myapp/secrets
++ KUBEFLOW_K8S_MANIFESTS_DIR=/kubeflow_dir/myapp/k8s_specs
++ KUBEFLOW_K8S_CONTEXT=myapp
++ ZONE=us-central1-a
++ EMAIL=formoso@google.com
++ KUBEFLOW_IP_NAME=myapp-ip
++ KUBEFLOW_ENDPOINT_NAME=myapp
++ KUBEFLOW_HOSTNAME=myapp.endpoints.data-science-on-kubs.cloud.goog
++ CONFIG_FILE=cluster-kubeflow.yaml
++ PROJECT_NUMBER=1067425295876
++ CLUSTER_VERSION=1.10.9-gke.0
+ '[' apply == generate ']'
+ '[' apply == apply ']'
+ '[' platform == platform ']'
+ '[' gcp == gcp ']'
+ updateDM
+ updateDeployment myapp cluster-kubeflow.yaml
+ local NAME=myapp
+ local CONFIG_FILE=cluster-kubeflow.yaml
+ pushd .
/kubeflow_dir/myapp /kubeflow_dir/myapp
+ cd /kubeflow_dir/myapp/gcp_config
+ set +e
++ gcloud deployment-manager --project=data-science-on-kubs deployments describe myapp
+ O='---
fingerprint: aaz3GBVCZg4lqxdt-IPcEQ==
id: '\''1851370381300158677'\''
insertTime: '\''2018-10-10T23:31:22.395-07:00'\''
name: myapp
operation:
  endTime: '\''2018-10-10T23:34:01.890-07:00'\''
  error:
    errors:
    - code: RESOURCE_ERROR
      location: /deployments/myapp/resources/myapp-gpu-pool-v1
      message: '\''{"ResourceType":"container.v1.nodePool","ResourceErrorCode":"400","ResourceErrorMessage":{"code":400,"message":"Accelerator
        type \"nvidia-tesla-k80\" does not exist in zone australia-southeast1-a.","status":"INVALID_ARGUMENT","statusMessage":"Bad
        Request","requestPath":"https://container.googleapis.com/v1/projects/data-science-on-kubs/zones/australia-southeast1-a/clusters/myapp/nodePools","httpMethod":"POST"}}'\''
  name: operation-1539239574608-577ee24119880-0fae6a11-aecae3c6
  operationType: update
  progress: 100
  startTime: '\''2018-10-10T23:32:55.484-07:00'\''
  status: DONE
  user: formoso@google.com
update:
  manifest: https://www.googleapis.com/deploymentmanager/v2/projects/data-science-on-kubs/global/deployments/myapp/manifests/manifest-1539239574983
NAME               TYPE                      STATE      INTENT
myapp              container.v1.cluster      COMPLETED
myapp-admin        iam.v1.serviceAccount     COMPLETED
myapp-gpu-pool-v1  container.v1.nodePool     FAILED     CREATE_OR_ACQUIRE
myapp-ip           compute.v1.globalAddress  COMPLETED
myapp-user         iam.v1.serviceAccount     COMPLETED
myapp-vm           iam.v1.serviceAccount     COMPLETED'
+ exists=0
+ set -e
+ '[' 0 -eq 0 ']'
+ echo myapp exists
myapp exists
+ gcloud deployment-manager --project=data-science-on-kubs deployments update myapp --config=cluster-kubeflow.yaml
ERROR: (gcloud.deployment-manager.deployments.update) Unable to read file 'cluster-kubeflow.yaml'.
root@google_sh:/kubeflow_dir/myapp#```

**Comment by jlewi** [comment]
@ashahba Which version of kfctl.sh is needed to pass zone as a command line flag? Is that supported in 0.3.1 or is a newer version needed?

**Comment by jlewi** [comment]
Closing this because I think its fixed by @ashahba's fixed in 0.4
